app/sentinel/cmd/main.go

```diff
@@ -185,7 +188,22 @@ func main() {
 		return
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGTERM, syscall.SIGINT)
+
+	go func() {
+		select {
```

Comment:
Just a doubt: Shouldn't this be in a for loop? Currently, the code handles the cancellation of the context only once, which means that subsequent signals received may not trigger further cancellation and could potentially leave the code running.